.github/workflows/bats/summarize.mjs

```diff
@@ -0,0 +1,340 @@
+// This file creates the summary table at the end of the run.
+//
+// Inputs:
+//   */version.txt   -- The version of Rancher Desktop tested
+//   */name.txt      -- The test suite that was ran
+//   */os.txt        -- The OS the test was run on
+//   */engine.txt    -- The container engine used
+//   */log-name.txt  -- The name of the logs artifact
+//   */report.tap    -- The results
+// Environment:
+//   GITHUB_API_URL, GITHUB_RUN_ID, GITHUB_REPOSITORY, GITHUB_SERVER_URL
+//     See https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables
+//   GITHUB_TOKEN
+//     GitHub authorization token.
+
+// @ts-check
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Define interface for emitting one line of output.
+ * @typedef {(line: string) => unknown} OutputMethod
+ */
+
+class Run {
+  /** Contents of version.txt. */
+  versionData = '';
+  /** Contents of name.txt. */
+  name = '';
+  /** Contents of os.txt. */
+  os = '';
+  /** Contents of engine.txt. */
+  engine = '';
+  /** Contents of log-name.txt. */
+  logName = '';
+  /** Total number of tests. */
+  total = 0;
+  /** Number of tests passed (not skipped). */
+  passed = 0;
+  /** Number of tests skipped. */
+  skipped = 0;
+  /** Number of tests failed. */
+  failed = 0;
+  /** Job ID; this may not be set. */
+  id = 0;
+  /** ID for the logs artifact; might be missing. */
+  logId = 0;
+  /** Number of tests passed or skipped. */
+  get ok() { return this.passed + this.skipped };
+  /** Whether this run succeeded. */
+  get succeeded() { return this.ok == this.total };
+  /** Version string for this run. */
+  get version() {
+    let v = this.versionData;
+    for (const prefix of ['Rancher Desktop-', 'rancher-desktop-']) {
+      if (v.startsWith(prefix)) {
+        v = v.substring(prefix.length);
+      }
+    }
+    for (const platform of ['linux', 'arm64-mac', 'mac', 'win']) {
+      const suffix = `-${ platform }.zip`;
+      if (v.endsWith(suffix)) {
+        v = v.substring(0, v.length - suffix.length);
+      }
+    }
+    return v;
+  }
+  /** The column for this run. */
+  get column() { return `${ this.os } ${ this.engine }` }
+
+  /**
+   * Compare two runs for sorting.
+   * @param {Run} left
+   * @param {Run} right
+   * @returns {number} -1, 0, or 1 for left {<|==|>} right.
+   */
+  static compare(left, right) {
+    return left.name.localeCompare(right.name) ||
+      left.os.localeCompare(right.os) ||
+      left.engine.localeCompare(right.engine);
+  }
+}
+
+/**
+ * Read the runs in the current directory.
+ * @returns {Promise<Run[]>}
+ */
+async function readRuns() {
+  /** @type Run[] */
+  const runs = [];
+
+  for (const entry of await fs.promises.readdir('.', { withFileTypes: true })) {
+    if (!entry.isDirectory()) {
```

Comment:
No, and even if it did it doesn't matter (those directories won't have the `report.tap`, etc. files, and we ignore errors reading any directories that don't look right, because we end up with `.github` etc.)